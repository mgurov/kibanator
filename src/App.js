import React from 'react';
import './App.css';
import AddWatchContainer from './components/AddWatchContainer.js'
import WatchListContainer from './components/WatchListContainer.js'
import DataListContainer from './components/datalist/DataListContainer.js'
import {ConfigContainer} from './components/config'
import Reset from './components/timerange/Reset'
import DocumentTitleContainer from './components/title/DocumentTitleContainer'

let App = () => {

  let showWatches = false

  let watches
  if (showWatches) {
    watches = (<div><WatchListContainer />
               <AddWatchContainer /></div>)
  }

  return (
    <div className="App">
      <DocumentTitleContainer />
      <div className="App-header">
        <ConfigContainer />&nbsp;<Reset />
      </div>
      {watches}
      <DataListContainer />
    </div>
  );
}

export default App;
