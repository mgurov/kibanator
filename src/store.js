import {
    createStore,
    applyMiddleware
} from 'redux'

import thunkMiddleware from 'redux-thunk'
import {readConfigFromLocalStore, writeConfigToLocalStore} from './reducers/config'
//import { createLogger } from 'redux-logger'
import {watchIndexData} from './state/data'

import kibanatorApp from './reducers'

const onNewHitsArrivedMiddleware = store => {
    return next => action => {
        
        let r = next(action)

        if (
            action.type === 'ACK_ID'
            || action.type === 'ACK_TILL_ID'
            || action.type === 'ACK_TAG'
            || action.type === 'ACK_ALL'
            || action.type === 'ADD_CAPTOR'
            || action.type === 'REMOVE_CAPTOR'
            || action.type === 'APPLY_DRAFT_FILTER'
            || action.type === 'ACK_PREDICATE'
        ) {
            store.dispatch({
                type: 'NEW_IDS_ARRIVED', //TODO: different type, also just listen for the changes
            })
            return r
        }

        if (action.type !== 'INCOMING_HITS') {
            return r
        }

        let state = store.getState();
        let watchIndex = action.payload.watchIndex
        let data = watchIndexData(state, watchIndex)
        if (data === undefined) {
            console.warn('no selected data to propagate incoming hits')
            return r
        }
        const newIds = data.hits.newIds

        if (newIds && newIds.length > 0) {
            store.dispatch({
                type: 'NEW_IDS_ARRIVED',
                payload: {
                    newIds,
                    watchIndex,
                },
            })
        }
        return r
    }
}

export function newStore() {
    let store = createStore(
        kibanatorApp,
        {config: readConfigFromLocalStore()},
        applyMiddleware(thunkMiddleware, onNewHitsArrivedMiddleware)
    )
    store.dispatch({
        type: 'ON_INIT'
    })
    let [currentConfig] = [null]
    store.subscribe(() => {
        let state = store.getState()
        if (currentConfig !== state.config) {
            writeConfigToLocalStore(state.config)
            currentConfig = state.config
        }
    })
    return store
}