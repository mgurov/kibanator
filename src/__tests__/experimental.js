import React from 'react';
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import fetchMock from 'fetch-mock'

import App from '../App';
import {newStore} from '../store'

afterEach(() => {
  fetchMock.restore()
});

xtest('snapshot testing example', () => {
  const rendered = <Provider store={newStore()}><App /></Provider>
  const component = renderer.create(rendered);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('clicking at the first fetch option changes the controls shown', () => {

  const rendered = <Provider store={newStore()}><App /></Provider>
  const wrapper = mount(rendered);

  expect(wrapper.find('SyncTimeControl').length).toEqual(0)
  expect(wrapper.find('SelectTimeRange').length).toEqual(1)

  fetchMock.mock('*', 401)
  let firstTimerangeButton = wrapper.find('button.preselected-time-range').first()
  firstTimerangeButton.simulate('click')

  expect(wrapper.find('SyncTimeControl').length).toEqual(1)
  expect(wrapper.find('SelectTimeRange').length).toEqual(0)
});