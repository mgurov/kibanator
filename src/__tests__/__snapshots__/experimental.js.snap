// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing example 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <span>
      v
      DEV
    </span>
       
    <button
      className="btn btn-xs glyphicon glyphicon-wrench btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    />
     
    <button
      className="btn btn-xs glyphicon glyphicon-repeat btn btn-default"
      disabled={true}
      onClick={[Function]}
      title="Reset"
      type="button"
    />
  </div>
  <div>
    <div
      className="btn-group"
      title="Sync"
    >
      <button
        className="preselected-time-range btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        15 mins
      </button>
      <button
        className="preselected-time-range btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        1 hour
      </button>
      <button
        className="preselected-time-range btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        1 day
      </button>
      <button
        className="preselected-time-range btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        1 week
      </button>
      <button
        className="preselected-time-range btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        1 month
      </button>
      <button
        className="btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Custom...
      </button>
    </div>
    <div
      className="container-fluid"
    />
    <div
      className="container-fluid"
    />
  </div>
</div>
`;
